src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -599,9 +599,9 @@ trait Implicits {
                 false
               } else {
                 val targs = solvedTypes(tvars, allUndetparams, allUndetparams map varianceInType(wildPt), upper = false, lubDepth(tpInstantiated :: wildPt :: Nil))
-                val AdjustedTypeArgs(okParams, okArgs) = adjustTypeArgs(allUndetparams, tvars, targs)
-                val remainingUndet = allUndetparams diff okParams
-                val tpSubst = deriveTypeWithWildcards(remainingUndet)(tp.instantiateTypeParams(okParams, okArgs))
+                val adjusted = adjustTypeArgs(allUndetparams, tvars, targs)
```

Comment:
At this point, only two of the four lists in the `AdjustedTypeArgs` are used. Would it be worth it to add an extra parameter, to indicate which lists are needed or not?

Reply:
I refactored this further to use `adjusted.undetParams` rather than recalculating.

It's only `allArgs` that isn't used at each call site, it is only used in `inferMethodInstance`.

We could refactor further to avoid paying for that when its not needed, but it will complicate the code somewhat and I'm not sure its worth it.